File: dynamic-pricing-business-rules-engine/backend/app/schemas/rule_testing.py
from decimal import Decimal

from pydantic import BaseModel, Field


class RuleTestRequest(BaseModel):
    rule_id: int = Field(..., ge=1)
    customer_type: str | None = None
    customer_category: str | None = None
    location: str | None = None
    quantity: int = Field(..., ge=1)
    category_id: int = Field(..., ge=1)
    product_id: int = Field(..., ge=1)
    base_price: Decimal = Field(..., ge=0)


class RuleTestResponse(BaseModel):
    rule_id: int
    rule_name: str
    matched: bool
    execution_type: str
    message: str
    discount_amount: Decimal

File: dynamic-pricing-business-rules-engine/frontend/src/types/ruleTesting.ts
export interface RuleTestRequest {
  rule_id: number;
  customer_type?: string | null;
  customer_category?: string | null;
  location?: string | null;
  quantity: number;
  category_id: number;
  product_id: number;
  base_price: number;
}

export interface RuleTestResponse {
  rule_id: number;
  rule_name: string;
  matched: boolean;
  execution_type: string;
  message: string;
  discount_amount: number;
}